lib/matplotlib/tight_layout.py

```diff
@@ -278,41 +278,22 @@ def get_tight_layout_figure(fig, axes_list, subplotspec_list, renderer,
         None if tight_layout could not be accomplished.
     """
 
-    subplot_list = []
-    nrows_list = []
-    ncols_list = []
-    ax_bbox_list = []
-
-    # Multiple axes can share same subplot_interface (e.g., axes_grid1); thus
-    # we need to join them together.
-    subplot_dict = {}
-
-    subplotspec_list2 = []
-
-    for ax, subplotspec in zip(axes_list, subplotspec_list):
-        if subplotspec is None:
-            continue
-
-        subplots = subplot_dict.setdefault(subplotspec, [])
-
-        if not subplots:
-            myrows, mycols, _, _ = subplotspec.get_geometry()
-            nrows_list.append(myrows)
-            ncols_list.append(mycols)
-            subplotspec_list2.append(subplotspec)
-            subplot_list.append(subplots)
-            ax_bbox_list.append(subplotspec.get_position(fig))
-
-        subplots.append(ax)
-
-    if len(nrows_list) == 0 or len(ncols_list) == 0:
+    # Multiple axes can share same subplotspec (e.g., if using axes_grid1);
+    # we need to group them together.
+    ss_to_subplots = {ss: [] for ss in subplotspec_list}
+    for ax, ss in zip(axes_list, subplotspec_list):
+        ss_to_subplots[ss].append(ax)
+    ss_to_subplots.pop(None, None)  # Skip subplotspec == None.
+    if not ss_to_subplots:
         return {}
+    subplot_list = list(ss_to_subplots.values())
+    ax_bbox_list = [ss.get_position(fig) for ss in ss_to_subplots]
```

Comment:
Now `_auto_adjust_subplotpars` is always called with `ax_bbox_list != None`, so maybe it should take a `dict` instead of these two lists, and move `None` handling to the deprecated wrapper?

Reply:
Actually, I was thinking that after the deprecation period, I could just delete the ax_bbox_list kwarg, stop computing it on the call side, and just always compute it within _auto_adjust_subplotpars?